RATIONALE.md

```diff
@@ -13,21 +13,41 @@ Note: `microwasm` was never specified formally, and only exists in a historical
 https://github.com/bytecodealliance/wasmtime/blob/v0.29.0/crates/lightbeam/src/microwasm.rs
 
 
-## Size limitations
-### Number of functions
+## Implementation limitations
+
+[WebAssembly specification states that](https://www.w3.org/TR/wasm-core-1/#a2-implementation-limitations)
+runtime implementors can impose their own limits on a number of aspects of a Wasm module or execution.
+
+The followings are the limitations we explicitly impose and handle as errors during module instantiation in wazero:
+
+### Number of functions in a store
 
 The possible number of function instances in [a store](https://www.w3.org/TR/wasm-core-1/#store%E2%91%A0) is not specified in the WebAssembly specifications since [`funcaddr`](https://www.w3.org/TR/wasm-core-1/#syntax-funcaddr) corresponding to a function instance can be arbitrary number. 
-In wazero, we choose to use `uint32` to represent `funcaddr`. Therefore the maximum number of function instances a store can instantiate is limited to 2^32. 
+In wazero, we choose to set the maximum number of function instances a store can instantiate is limited to 2^27 as that would result in occupying 2^20 (= 1 GiB) bytes in memory for function pointers alone.
```

Comment:
wazero limits the maximum function instances to 2^27 as even that number would occupy 1GB in function pointers